### PR TITLE
Added Navigation 

### PR DESCRIPTION
**Pull request by r4z396**

### CODE REVIEW

File: GUI/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required (VERSION 3.8)
 
 file(GLOB_RECURSE sources CONFIGURE_DEPENDS "*.cpp" "*.h")
-add_executable (GUI ${sources})
+add_executable (GUI ${sources} "include/RTSUnitLayer.h" "src/RTSUnityLayer.cpp" "include/StatsSummary.h" "src/StatsSummary.cpp")
```
Comment (CommanderCero) [inline GUI/CMakeLists.txt:4]: Remove

File: GUI/include/GridUtils.h [hunk 2]
```diff
@@ -43,6 +43,15 @@ static sf::Vector2f toGridFloat(sf::Vector2f vec)
 	return sf::Vector2f((x), (y));
 }
 
+static SGA::Vector2f toGridSGAFloat(sf::Vector2f vec)
```
Comment (CommanderCero) [inline GUI/include/GridUtils.h:46]: What is this needed for

File: GUI/CMakeLists.txt (hunk 1 above)
Comment (r4z396) [reply GUI/CMakeLists.txt]: done

File: GUI/include/GridUtils.h (hunk 2 above)
Comment (r4z396) [reply GUI/include/GridUtils.h]: Not necessary(Removed)

File: GUI/src/RTSOverlayLayer.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	  const dtNavMesh* mesh = state.navigation->m_navMesh;
+
+	  if (mesh)
+	  {
+		  //Draw navmesh
+		  for (int i = 0; i < state.navigation->m_navMesh->getMaxTiles(); ++i)
+		  {
+			  const dtMeshTile* tile = mesh->getTile(i);
+
+			  if (!tile->header) continue;
```
Comment (CommanderCero) [inline GUI/src/RTSOverlayLayer.cpp:31]: No

File: Stratega/include/ForwardModel/Action.h [hunk 4]
```diff
@@ -27,6 +27,9 @@ namespace SGA
 		const T& getTargetPosition() const { return targetPosition; }
 		const int getPlayerID() const { return playerID; }
 
+		void updateTargetPosition(T& target) { targetPosition = target; }
```
Comment (CommanderCero) [inline Stratega/include/ForwardModel/Action.h:30]: dead code

File: Stratega/include/Representation/Navigation.h [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+{
+	SAMPLE_PARTITION_WATERSHED,
+	SAMPLE_PARTITION_MONOTONE,
+	SAMPLE_PARTITION_LAYERS,
+};
+
+static const int MAX_POLYS = 256;
+
+// Returns a random number [0..1]
+static float frand()
```
Comment (CommanderCero) [inline Stratega/include/Representation/Navigation.h:31]: Not needed

File: Stratega/include/Representation/RTSGameState.h [hunk 6]
```diff
@@ -6,11 +6,13 @@
 #include <Representation/UnitType.h>
 #include <Representation/Player.h>
 
+#include "Navigation.h"
```
Comment (CommanderCero) [inline Stratega/include/Representation/RTSGameState.h:9]: Relative include

File: Stratega/include/Representation/RTSGameState.h [hunk 7]
```diff
@@ -36,11 +40,13 @@ namespace SGA
 			lastUsedPlayerID(-1),
 			lastUsedUnitID(-1),
 			winnerPlayerID(-1),
-			tileScale(1)
-		{
-		}
+			tileScale(1),
+			navigation( std::make_shared<Navigation>())
```
Comment (CommanderCero) [inline Stratega/include/Representation/RTSGameState.h:44]: cant this be a nullptr?

**Pull request merged**